src/Model/Subscription.php

```diff
@@ -303,24 +309,34 @@ public function getPaymentMethod(): ?PaymentMethod
     /**
      * @return Subscription
      */
-    public function setPaymentMethod(?PaymentMethod $paymentMethod): self
+    public function setPaymentMethod(PaymentMethod $paymentMethod): self
     {
         $this->paymentMethod = $paymentMethod;
 
         return $this;
     }
 
-    public function getPlan(): ?Plan
+    public function getPlan(): Plan
     {
         return $this->plan;
     }
 
     /**
      * @return Subscription
      */
-    public function setPlan(?Plan $plan): self
+    public function resetPlan(Plan $plan): self
+    {
+        $this->plan = new NullPlan();
+
+        return $this->setPlan($plan);
```

Comment:
not sure I'm catching why we setting the plan on 329 then resetting it on 331? or is that to clear out the plan props... would `$this->setPlan(new NullPlan())` be better on 329

Reply:
Probably a good idea to call setPlan twice. It shall be done.